source/extensions/path/uri_template_lib/uri_template.h

```diff
@@ -29,6 +28,10 @@ struct ParsedSegment {
   RewriteStringKind kind_;
 };
 
+struct RewriteSegments {
```

Comment:
nit: Please comment this struct with a description of what it contains.

Actually, I think it would be better to defined this type as

```
using RewriteSegment = absl::variant<int, std::string>>;
using RewriteSegments = std::vector<RewriteSegment>;
```
This will avoid some unnecessary ".segments" in various places. And will help in some test boilerplate later on.

Reply:
Sure. Thought about doing this when I was changing. Happy to see we're on the same page.. Will change